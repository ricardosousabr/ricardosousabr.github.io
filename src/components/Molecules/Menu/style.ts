import styled from "styled-components";
import screens from "../../../tokens/screen";


export const MenuStyle = styled.div`
display: flex;
flex-direction: column;
`;

export const BoxMenu = styled.div`
  display: flex;
  justify-content: flex-end;
  padding-right: 16px;
  padding-top: 16px;

  @media (min-width: ${screens.small}) {
    display: none;
  }
`
export const MenuMobile = styled.div`

  @media (min-width: ${screens.small}) {
    display: none;
  }
`

export const MenuDesktop = styled.div`
  display: none;

  @media (min-width: ${screens.small}) {
    display: block;
  }
`
