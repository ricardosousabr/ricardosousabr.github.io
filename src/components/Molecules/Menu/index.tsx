import { useState } from "react"
import List from "./components/List"
import Button from "../../Atoms/Button";
import { MenuStyle, BoxMenu, MenuMobile, MenuDesktop } from "./style"

export default function Menu() {
  const menuItens = ["About", "Skils", "Experience", "Education", "Contact", "Switch Theme", "Download CV"];
  const [valueButton, setValueButton] = useState<boolean>(true)

  function displayMenu() {
    return setValueButton(!valueButton)
  }

  return (
    <>
      <MenuStyle>
        <BoxMenu>
          <Button onClick={() => displayMenu()} background="" padding="10px 20px" borderRadius="10px" color="black" border="none">
            <svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" width="15" height="15" viewBox="0 0 50 50">
              <path d="M 3 9 A 1.0001 1.0001 0 1 0 3 11 L 47 11 A 1.0001 1.0001 0 1 0 47 9 L 3 9 z M 3 24 A 1.0001 1.0001 0 1 0 3 26 L 47 26 A 1.0001 1.0001 0 1 0 47 24 L 3 24 z M 3 39 A 1.0001 1.0001 0 1 0 3 41 L 47 41 A 1.0001 1.0001 0 1 0 47 39 L 3 39 z"></path>
            </svg>
          </Button>
        </BoxMenu>
        {
          valueButton ? <MenuMobile><List menuItens={menuItens} /></MenuMobile> : ""
        }
        <MenuDesktop><List menuItens={menuItens} /></MenuDesktop>
      </MenuStyle>
    </>
  )
}
