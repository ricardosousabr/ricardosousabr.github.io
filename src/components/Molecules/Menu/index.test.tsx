import renderer from "react-test-renderer";
import Menu from "./index";

test("Link renders correctly", () => {
  const menuItens = ["About", "Skils", "Experience", "Education", "Contact", "Switch Theme", "Download CV"]
  const tree = renderer.create(<Menu menuItens={menuItens} />).toJSON();
  expect(tree).toMatchSnapshot();
});
