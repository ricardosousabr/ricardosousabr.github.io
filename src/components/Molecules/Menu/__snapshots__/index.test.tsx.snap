// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-dlniIP fPfzoH"
>
  <div
    className="sc-hKFymg fMogKo"
  >
    <button
      background="transparent"
      border="none"
      borderRadius="10px"
      className="sc-bdnyFh eSbMzv"
      color="black"
      onClick={[Function]}
      padding="10px"
    >
      <svg
        fill="none"
        height="32"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 12H20"
          stroke="#D1D5DB"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M4 6H20"
          stroke="#D1D5DB"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M4 18H20"
          stroke="#D1D5DB"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-jSFipO gLzvjD"
  >
    <ul
      className="sc-gtsqUy chnRqi"
    >
      <li>
        <button
          background="transparent"
          border="none"
          borderRadius="8px"
          className="sc-bdnyFh lfCnzj"
          color="white"
          fontSize="16px"
          padding="8px 12px"
          type="button"
        >
          About
        </button>
      </li>
      <li>
        <button
          background="transparent"
          border="none"
          borderRadius="8px"
          className="sc-bdnyFh lfCnzj"
          color="white"
          fontSize="16px"
          padding="8px 12px"
          type="button"
        >
          Skils
        </button>
      </li>
      <li>
        <button
          background="transparent"
          border="none"
          borderRadius="8px"
          className="sc-bdnyFh lfCnzj"
          color="white"
          fontSize="16px"
          padding="8px 12px"
          type="button"
        >
          Education
        </button>
      </li>
      <li>
        <button
          background="transparent"
          border="none"
          borderRadius="8px"
          className="sc-bdnyFh lfCnzj"
          color="white"
          fontSize="16px"
          padding="8px 12px"
          type="button"
        >
          Contact
        </button>
      </li>
    </ul>
  </div>
</div>
`;
